MODEL_CONTRIBUTION_GUIDE.md

```diff
@@ -0,0 +1,218 @@
+# Model Contribution Guide
```

Comment:
silly nit, but we have `CONTRIBUTING.md`, should we make this `CONTRIBUTING_MODELS.md` or something to keep the filename parallel?